File: Java/com/manish/assignments/four/Subject.java
package com.manish.assignments.four;

public class Subject {
	
	private int marks;
	private Grade grade;
	private	String name;
	
	Subject(String name, int marks){
		this.name=name;
		this.marks=marks;
		this.grade=Grade.F;
		if(this.marks>=60) {
			this.grade=Grade.C;
			if(this.marks>=70) {
				this.grade=Grade.B;
				if(this.marks>=80) {
					this.grade=Grade.A;
					if(this.marks>=90) {
						this.grade=Grade.Ex;
					}
				}
			}
		}
	}
	
	public String getName() {return this.name;}
	public int getMarks() {return this.marks;}
	public Grade getGrade() {return this.grade;}
}

File: CPP/4/src/Subject.cpp
/*
 * Subject->cpp
 *
 *  Created on: 18-Aug-2024
 *      Author: Manish Patil
 */

#include "Subject.h"

Subject::Subject(string name,int marks) {
	this->name=name;
	this->marks=marks;
	if(this->marks>=60) {
				this->grade="C";
				if(this->marks>=70) {
					this->grade="B";
					if(this->marks>=80) {
						this->grade="A";
						if(this->marks>=90) {
							this->grade="Ex";
						}
					}
				}
			}
}

string Subject::getName(){return this->name;}
int Subject::getMarks(){return this->marks;}
string Subject::getGrade(){return this->grade;}



